plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraMetadata.java

```diff
@@ -203,29 +209,51 @@ public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, Conn
         return columnHandles.buildOrThrow();
     }
 
-    @SuppressWarnings("deprecation") // TODO Implement streamTableColumns method
     @Override
-    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    public Iterator<RelationColumnsMetadata> streamRelationColumns(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
     {
-        requireNonNull(prefix, "prefix is null");
-        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
-        for (SchemaTableName tableName : listTables(session, prefix)) {
+        Map<SchemaTableName, RelationColumnsMetadata> relationColumns = new HashMap<>();
+
+        SchemaTablePrefix prefix = schemaName.map(SchemaTablePrefix::new)
```

Comment:
it would be much more readable if we had a SchemaTablePrefix(Optional<String> schema) constructor. At first glance I have no idea what's happening here because two different constructors are involved.

`.orElse(new SchemaTablePrefix());` would be better I think.